pdf/model/textencoding/simple.go

```diff
@@ -0,0 +1,1735 @@
+	}
+	return glyph, ok
+}
+
+// GlyphToCharcode returns character code for glyph `glyph`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (se SimpleEncoder) GlyphToCharcode(glyph string) (uint16, bool) {
+	code, ok := se.glyphToCode[glyph]
+	if !ok {
+		common.Log.Debug("Glyph -> Charcode error: glyph not found: %q", glyph)
+	}
+	return code, ok
+}
+
+// RuneToCharcode returns the PDF character code corresponding to rune `r`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (se SimpleEncoder) RuneToCharcode(val rune) (uint16, bool) {
+	return doRuneToCharcode(se, val)
+}
+
+// CharcodeToRune returns the rune corresponding to character code `code`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (se SimpleEncoder) CharcodeToRune(code uint16) (rune, bool) {
+	r, ok := se.codeToRune[code]
+	if !ok {
+		common.Log.Debug("Charcode -> Rune error: charcode not found: 0x%04x", code)
+	}
+	return r, ok
+}
+
+// RuneToGlyph returns the glyph corresponding to rune `r`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (se SimpleEncoder) RuneToGlyph(r rune) (string, bool) {
+	return runeToGlyph(r, glyphlistRuneToGlyphMap)
+}
+
+// GlyphToRune returns the rune corresponding to glyph `glyph`.
+// The bool return flag is true if there was a match, and false otherwise.
+func (se SimpleEncoder) GlyphToRune(glyph string) (rune, bool) {
+	return glyphToRune(glyph, glyphlistGlyphToRuneMap)
+}
+
+// ToPdfObject returns `se` as a PdfObject
+func (se SimpleEncoder) ToPdfObject() core.PdfObject {
+	if se.differences == nil {
+		return core.MakeName(se.baseName)
+	}
+	dict := core.MakeDict()
+	dict.Set("Type", core.MakeName("Encoding"))
+	dict.Set("BaseEncoding", core.MakeName(se.baseName))
+	dict.Set("Differences", core.MakeArray(ToFontDifferences(se.differences)...))
+	return core.MakeIndirectObject(dict)
+}
+
+// makeEncoder returns a SimpleEncoder based on `codeToRune`.
+func (se *SimpleEncoder) makeEncoder() {
+	codeToRune := map[uint16]rune{}
+	for code, r := range se.baseEncoding {
+		codeToRune[code] = r
+	}
+	if se.differences != nil {
+		for code, glyph := range se.differences {
+			r, ok := GlyphToRune(glyph)
+			if !ok {
+				common.Log.Debug("ERROR: No match for glyph=%q differences=%+v", glyph, se.differences)
+			}
+			codeToRune[uint16(code)] = r
+		}
+	}
+
+	codeToGlyph := map[uint16]string{}
+	glyphToCode := map[string]uint16{}
+	for code, r := range codeToRune {
+		glyph := glyphlistRuneToGlyphMap[r] // !@#$ Build out this map
+		codeToGlyph[code] = glyph
+		glyphToCode[glyph] = code
+		if glyph == "" {
+			common.Log.Debug("ERROR: Empty glyph code=0x%04x r=%+q=%#q", code, r, r)
+		}
+	}
+	se.codeToGlyph = codeToGlyph
+	se.glyphToCode = glyphToCode
+	se.codeToRune = codeToRune // XXX: !@#$ Make this a string
+}
+
+func FromFontDifferences(diffList []core.PdfObject) (map[byte]string, error) {
```

Comment:
godoc comment
// FromFontDifferences loads a Differences list and returns a map of byte to string representing character code to glyph mappings.

Reply:
Done